### ISSUE TITLE
confusing UI + do I need to have the specific item page open?

### ISSUE THREAD

**Issue by spiderbit** [sweitzel/Biet-O-Matic-BE#28]
So I have some problems comments and questions about this addon:
1. The UI is not intuitive I have a college degree for applied computer sciences yet I failed to understand it completely for a long time (most articles I was overbitten anyway so I did not see it fail often therefor I did not notice for a long time that I did something wrong), sure partially that is because I did not read the readme careful but show me any other browser addon that you need to.
2. to be more specific there is a lot questionable with that UI but the biggest one I failed yesterday is the [ ] inactive behind the items, I thought if group bid is active that would be enough and that you can "inactivate" then some items if you just want to look at it but not bid on it. That would make more sense to me. But small fix to make it more intuitive would be to change the tex from [ ] inactive to [ ] activate -> activated. If I have a checkpoint I expect to read behind the action the button does not what status the button is because the status is already visible without a cross it's "deactivated" so the implied text is "deactivated inactive" which is very confusing. So maybe change the text to "activate" or just keep the text "active" then I see the trigger on active is off so it's not active.
3. tomorrow even I had all 3 autobid buttons active global / group and item specific it did not bid again, the time window was 8 seconds my bid price was 20 euro over the highest bid, the overview window was open and in foreground, the only suspicion what I might done wrong, but which is not listed in the readme, do I have to have the specific article window open? I see no log and therefor it does not tell me why it did not bid, just that the bidding is over.

I find even the other 2 toggles not to intuitive but you can figure it out because the changing text is in the button but having a toggle field with the text "inactive" behind is really confusing and I never seen such a UI design in any other app or extension.

4. ohh and I forgot the debug/test bidding option seem to not be available anymore or I am to dumb to start it, don't see anything in the readme?

**Your Environment (please complete the following information):**
 - OS: Nixos
 - Browser Name and Version : Firefox 84.0.1
 - Bid-O-Matic BE Version: 0.5.7

**Comment by sweitzel** [comment]
Thanks for your feedback. The recommendation to read the instructions is explicitly mentioned in the Addon Store. I had the idea to add a "tutorial/wizard", but as this is a hobby project I decided put it to the very end of the todo list. but I am still looking for volunteers to help with implementing improvements.

1. I spent a lot of time to create documentation. I expect from users to read at least the quick start guide.
2. Agree, that can be improved to make it more intuitive, if BE would bid for an item or not.
3. Did you check the Item log under `+`?  The item page doesn't need to be open, BE will open a bid page (as documented) If there is no specific reason in the item log why it didn't bid, it probably means that the extension was not active at that time. This could be for example if the overview tab was put into "sleep mode", which can sometimes happen if there is a specific extension installed (Great Suspender) or the browser itself can do it. The overview tab is the orchestrator, which initiates the bidding process. Its bases on javascript timers. If the timers are not called, bidding cannot happen.
4. Test mode is still there, but in Firefox you need to shift-click on the BE icon in the toolbar, instead of the overview page Auto-Bid button.

I see your requests mostly as improvements. If there is a bug which generally prevents the usability of this extension, of course this gets higher priority. I do not have currently much time to work on this project, so feel free to contribute to this Open Source project via a PR submission.

**Comment by mikhoul** [comment]
I agree that the UI is poor and could be a lot better, more intuitive **BU**T at the same time I understand that it is a project that is developed as a hobby so it's hard sometime to prioritize what to do first with the free time availlable..

When I'm coding a GUI lot of time the first revision is poor because I know where are all the functions since it's me that creted the whole software. So I ask to a friend that don,t know anything about the program to try it to see where are the flaws and improve it, make the  GUI more intuitive.

In the next months if I have some spare time myself I will try to make suggestions and some mockup UI  to help the project.

There is another eBay extension that can be useful too: https://powersniper.sigma-solutions.fr/ the dev like here is also very open to comments by email BUT it is not open source sadly but personally I trust the dev and it's easy to read the source.

Both BOM and PS have their strength and weakness so I use both (not at the same time :smiley_cat:  .)

**Comment by spiderbit** [comment]
@sweitzel Ok that helped a lot got Test mode active

And found the debug log:
Opening offer tab now failed.

So I will try in the next bidding in 3 hours to open the tab manually that maybe could prevent that from happening. I have generally "no tabs" extension installed but it's deactivated so that should not be the issue.

**Comment by sweitzel** [comment]
> And found the debug log:
> Opening offer tab now failed.

Thats an internal error, never saw that and currently do not know what could cause this. It would be great to know why this happens. Unfortunately, the details are logged in regular Browser Console, under string ` Unable to open tab for article` along with details about the issue.

**Comment by spiderbit** [comment]
so in general it worked, in the end, will keep a eye on it, seems to sometimes work and sometimes not. But did snipe me a ryzen 2700x
